source/ota.c

```diff
@@ -2972,13 +2973,34 @@ void OTA_EventProcessingTask( void * pUnused )
      * OTA Agent is ready to receive and process events so update the state to ready.
      */
     otaAgent.state = OtaAgentStateReady;
+    otaAgent.agentStarted = true;
 
     while( otaAgent.state != OtaAgentStateStopped )
     {
         receiveAndProcessOtaEvent();
     }
 }
 
+OtaState_t OTA_EventProcess( void * pUnused )
+{
+    ( void ) pUnused;
+
+    /* Since user can single cycle the agent, we cannot repeatedly change state back to OtaAgentStateReady
+     * as it may interfere with an ongoing OTA. However, OtaAgentStateReady must still indicate agent readiness.*/
+    if( !otaAgent.agentStarted )
```

Comment:
Could we check for `(otaAgent.state == OtaAgentStateInit )` instead of adding new variable `agentStarted` ?
Are there other transitions to ready state, for example after stopping OTA can the agent be started again?

Reply:
I don’t think the checking for `state == OtaAgentStateInit` will work. `EventProcessingTask` sets state to `OtaAgentStateReady` irrespective of any conditions. From what I’ve seen, `OtaAgentStateReady` is strictly for indicating when ota thread has started running. To match that criteria we couldn’t check for state == `OtaAgentStateInit` since agent is initialized as `OtaAgentStateStopped` (see ota.c:498). So at first call to `OTA_EventProcess` in user OTA thread, `state != OtaAgentStateInit` unless user has called `OTA_Init` _before_ user ota daemon has had a chance to call `OTA_EventProcess`.

Reply:
IMO `OTA_EventProcess` should never be called before `OTA_Init()` as it sets the PAL functions and thing name etc.. in OTA Init.

Reply:
I agree, but it doesn't discount that you _can_ technically create the `OTA_EventProcessingTask` before calling `OTA_Init` and app would still work. Matching readiness based on this condition != conditions by which `OTA_EventProcessingTask` sets state to `OtaAgentStateReady`. 

Though the mentioned is an _incorrect_ flow, at least in that it exclusively produces error logs, it does _work_ since OTAs would work nominally after `OTA_Init` _is_ called.

Reply:
Regardless of that technical tidbit, I do like like the suggested change. Added it in the latest commit, with a comment stating that the readiness it indicates is only _approximately_ equal to readiness indicated by `OTA_EventProcessingTask`.